pysat/_instrument.py

```diff
@@ -788,20 +791,22 @@ def __getitem__(self, key):
             inst[datetime1:datetime2, 'name1':'name2']
 
         """
+        if data is None:
+            data = self.data
```

Comment:
When would you want to select from another data object, from meta? Or is there another case for this. (I like this change btw)

Reply:
The case that led to this change is in `load`.  When you pad data, you want to the previous and next days, select the appropriate time window, and then concat the data together.  Without this, you couldn't slice reliably without reassigning the data you want to slice to `self.data`.  Now you can!